src/allomancy/java/leaf/cosmere/allomancy/common/manifestation/AllomancyBendalloy.java

```diff
@@ -27,21 +30,59 @@ public AllomancyBendalloy(Metals.MetalType metalType)
 	@Override
 	protected void applyEffectTick(ISpiritweb data)
 	{
-		int mode = getMode(data);
-
 		if (AllomancyEntityThread.serverShutdown)
 		{
 			//don't start up new threads if the server is shutting down
 			return;
 		}
 
+		int mode = getMode(data);
 		String uuid = data.getLiving().getStringUUID();
-		if (mode > 0 && !playerThreadMap.containsKey(uuid))
+
+		// data thread management
 		{
-			playerThreadMap.put(uuid, new BendalloyThread(data));
+			if (mode > 0 && !playerThreadMap.containsKey(uuid))
+			{
+				playerThreadMap.put(uuid, new BendalloyThread(data));
+			}
+
+			playerThreadMap.entrySet().removeIf(entry -> !entry.getValue().isRunning || AllomancyEntityThread.serverShutdown || entry.getValue() == null);
 		}
 
-		playerThreadMap.entrySet().removeIf(entry -> !entry.getValue().isRunning || AllomancyEntityThread.serverShutdown);
+		// data processing
+		{
+			// check if bendalloy is off or compounding
+			if (mode <= 0)
+			{
+				return;
+			}
+
+			// this is the only way to check if the player is still online, thanks forge devs
+			if (data.getLiving().level.getServer().getPlayerList().getPlayer(data.getLiving().getUUID()) == null)
+			{
+				return;
+			}
+			//offload to power's tick
+			boolean isActiveTick = getActiveTick(data) % 20 == 0;
```

Comment:
it should always be active tick if they are in here as this it the applyEffectTick function

Reply:
Strangely, somehow, I think that's your code. I don't comment like that, so probably a remnant? Will remove either way :)

Reply:
yeah, sorry to make you clean up after me! 😅